### PR TITLE
[0.3.0] lockup: Add lockup options

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
- Introduced optional release duration
- Introduced optional absolute lockup timestamp.
- Updated `init` arguments
    - Added optional absolute `lockup_timestamp`.
    - Renamed `lockup_start_information` to `transfers_information`. Also renamed internal timestamp to `transfers_timestamp`.
    - Added optional `release_duration` to linearly release tokens.

## TODO:

- [x] Add README examples for absolute timestamp and release duration
- [ ] Add integration test for release duration

## Test plan:
- Added unit tests to cover absolute lockup timestamp and release duration.
- Existing unit/integration tests.

### CODE REVIEW

File: lockup/src/getters.rs [hunk 1]
```diff
@@ -61,12 +61,19 @@ impl LockupContract {
...
-                .0
-                .saturating_add(self.lockup_information.lockup_duration.0);
+            let lockup_timestamp = std::cmp::max(
+                transfers_timestamp
+                    .0
+                    .saturating_add(self.lockup_information.lockup_duration.0),
+                self.lockup_information
+                    .lockup_timestamp
+                    .map(|t| t.0)
+                    .unwrap_or(0),
```
Comment (bowenwang1996) [inline lockup/src/getters.rs:75]:
```suggestion
                self.lockup_information
                    .lockup_timestamp
                    .map_or(0, |t| t.0),
```

File: lockup/src/getters.rs [hunk 2]
```diff
@@ -61,12 +61,19 @@ impl LockupContract {
...
-        if let LockupStartInformation::TransfersEnabled { lockup_timestamp } =
-            &self.lockup_information.lockup_start_information
+        if let TransfersInformation::TransfersEnabled {
+            transfers_timestamp,
+        } = &self.lockup_information.transfers_information
         {
-            let lockup_timestamp = lockup_timestamp
-                .0
-                .saturating_add(self.lockup_information.lockup_duration.0);
+            let lockup_timestamp = std::cmp::max(
```
Comment (bowenwang1996) [inline lockup/src/getters.rs:68]: why is it a maximum of the two here? shouldn't it always be transfer_timestamp + lock duration?

File: lockup/README.md [hunk 3]
```diff
@@ -26,6 +29,13 @@ In addition to the lockup period that starts from the moment the transfers are e
 all tokens until `cliff_timestamp`.
 Once the `cliff_timestamp` passed, the tokens are vested on a pro rata basis from the `start_timestamp` to the `end_timestamp`.
 
+### Release schedule
+
+The release of tokens can also be scheduled to be linear from the moment transfers are enabled.
+To achieve this it's possible to specify release duration. Once the transfers are enabled, the release schedule will
+start from the timestamp of the vote.
+Release schedule can't be terminated by the foundation.
```
Comment (willemneal) [inline lockup/README.md:37]:
```suggestion
Release schedule can not be terminated by the foundation.
```

File: lockup/src/getters.rs (hunk 2 above)
Comment (evgenykuzyakov) [reply lockup/src/getters.rs:68]: We lock tokens to whichever is later. E.g. `40` days lock from the date of X or `14` days after the vote.

**Pull request merged**